routes/file.js

```diff
@@ -4,12 +4,25 @@ module.exports = async ( request, reply ) => {
 	const mime = require( 'mime-types' );
 	const minify = require( '../lib/minify' );
 	const compress = require( '../lib/compress' );
+	const validatePath = require( '../lib/validate-path' );
 	const { envBool } = require( '../lib/util' );
 	const { PerformanceObserver, performance } = require( 'perf_hooks' );
 
-	const path = request.query.path;
-
 	let log = {};
+
+	const userPath = request.query.path;
+
+	// Check user's path falls in the base path.
+	const path = validatePath( userPath );
+	if ( path === false ) {
```

Comment:
Nitpick: we can just do `!path` here, as we wouldn't want to do anything on any other falsy value anyway.

Reply:
Fixed in 08df020. I was initially scared of a really-edge-case of a file named "0", but upon checking, only php considers `"0"` to be false, javascript does not.